packages/uniforms-bridge-zod/src/ZodBridge.ts

```diff
@@ -28,6 +29,23 @@ function isNativeEnumValue(value: unknown) {
   return typeof value !== 'string';
 }
 
+function getLabel(value: unknown) {
+  return upperFirst(lowerCase(joinName(null, value).slice(-1)[0]));
+}
+
+function getFullLabel(path: ZodIssue['path'], indexes: number[] = []): string {
+  const lastElement = path[path.length - 1];
+
+  if (typeof lastElement === 'number') {
+    const slicedPath = path.slice(0, path.length - 1);
+    return getFullLabel(slicedPath, [lastElement, ...indexes]);
+  }
+
+  return indexes.length > 0
+    ? `${getLabel(path)} (${indexes.join(', ')})`
+    : getLabel(path);
+}
```

Comment:
I'm not sure about the format of the indexes that are printed.
Have a look at the [TS example](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAphYBDADnAskqwAWMC8MA9ANoB0JAOgLrVEDcAUKJLAsmgEpwDmcAHgWIAiIjyaMiAKikwAAgEswUOACcwSADYwpRRgDMArmGBQF4eIlRwAFNFVKeALhj3HAShgBvRjBiq4KEN1VygHMB4CfEIAchiYAB8Eyw4MLFwyFWg7MI9fPxgAfhgAAxJhABIvNwiyAJRNJGBbdmtuPn4AGhgYykphGPcAX2FqEvy-FxrxRiHGSRl5JRV1LR09IxMzC1a0ABUQACkQJRzw51Dz7qUAEwEXMEMAWwAjNU8fPxZoFOsboV2tmm7iYfgCQRCgP+0UI0yKMBstwE8JKZCq0yGJRgU1yEU8LihTDmzHAP2MgMw2DwhCIAD1KOUbGQpO5RhUGCTWDByVYuLxkTS+qIZgtZIplGoNNpdAZjKZzGBuWBAWdHDjzh98t9YE90tSlRS9WQBHBgKq8aD-IFgordVT4XbcCQAIzUOpwBpNWw81LtATdGIDfGXRxE+ZEIgwPY4BQQK08QyNVTwfgoAIQCAK1wNBRQONQHBwEMRGDLEAwAsBIsoJCqPNOSSRvwlNNwfQKfhY2MwJAwG4gKAqf4aJ5wbpwMg8MjdEogF4AK1NUDIyugcBuZAATGR23BNDcsVgK4XGwUz+f9KpwLAbIWTNWAu3+O4yDAAJKwUAANzUcaeIB+aAkDAG5a3+QwIFNJBIIgMhT2bCBDBeRAHBQKAuzjXsymEMhcNGLFEOQ4BUNgJB9BWY8HzbDtX2jIsWBUZR4PPFi-AAd1zGNFQgHAQETf43l+NABIAT0o7lINUcdJ2nUpyiFfC4IjAoSnTdDS0wsARILRwYH0EBNE0EAOJLAsi0IlCFDQ2jCxgTQ4CQG5dP7G9URKdxmNY7s3BQYSEVRJAsQAWgAPlKQLPAUfRKICDSYHACd31gbs4CeNCxKizyWJKCziKs9SUrSqARKUvRlIAQUM48qxgGs6zjZBFUE1L0oAQk5H4RzgP1BHwfI6RsEhaQoGhqCkGxCicSgKGGqhaAAancFkJqmshChsBlhBWlahtGBIhSWwohr6ah3EKUZ3Gm9bmXcM72QkZSAHV6OAsABwkosSnnE4wAAOSQUdDzjNiiwEFAQDrHs9LlbZFQ4gsewzBQeGVf5lklLRT0LTQ0FUTT-jM1wRNeAyYBwaCK3LICxMvOAi2MMxtDMsA4M2eULG+pR-tHN80s0Gx9EaC5HkM7pcLIOr6yVABrN62LAEhTvVRxFaYNnYZgTm-oBuBeYaJlcMliAXGMWXjIVpXixFZSEE0ZYgqciAkBeeyguVfgoCCu3lRgUA+YEXMxKCoKo1sv2GgD4r4ui3MYAAfVjOOyeRnBWZhrMte53W+YNiXayl025YtzU-GUt9osJ9s8dI1QE1HZQ4pKEXNBKbpmZgEGrQhRVgKE9ce1UVQkDEkBoqNuDS8jAB5MzVA4yDul7qErbi8EbXXV9yuBotdTE-8nP0MSW3ziAsUaFZOn6yMdLjcn83LQTcFNaX+7liffdJWA1-ULeAAUT6EEbEgAAGagURCDNwkJPGA3BEKaFgEbBEb02C8nXC+LUn8YBdWVhERWQhVbzGgdGbsjVkEwEEmEYCEB9KqFHGjZQ5YsI7ghgAUSaDgM+IAQAoB7CBcS9kIgI1HqWKAV9fa9zIYJUc-4FAAC91yL23jAfeUUj5GyBh3IsPByw0NqmoGhup7xWmgqSd+uibD2WSiBAUMBgH0FLNYwQrVoi1RPmQARPACz2PmvNJEz5vATA-lySWgCT4kD8dQKBBQKpVX0FoCAYkvxaEMAgFMzQ0KlGAYeWKyM3oBA3IEqKCIQlJFcZDGEtiS7niKTYYqaBhEhIqTEaYgwAmsWIJGAAyriHgmFYqOkLP8JAPAkBKB+ITAICYkwplbEjCwBYjwQBzMlKxUAQCBLPMpSsdMyl5gRFobQjVyGgyKiJdwDZ2nKWbK2J8GEoYuSHFgnW0kpwzjnIuUwK4EBDi3DuBQe4Dw9lgGZDZ0SmztL8Jea8CI7zNFqo+DsL4kofx-HjZRAFYBARAmBD6yAYLvxYlc0ouUSJ3KwuUcW+FXBIUshksiFFCY3JoqHei144BMUueCiFBR4ZcVcLxfixy+6iXEhBNQLzZLYQUmMAl54iWqW+WSrSOkSz6UMsZXShMSX5RskWeyjlnLvRsG5DynLuXeVyL5fuxqyCBRgKFcK7lSwV0LDkuMCVXwfjii1KOmUzUQpyjSvKaE7k+pKqCr4mCBlCC6j1Y0-BTQ2EliCCNzqEQDKqe07UMASDdCZV0alRESLdDUmAwgAyontJqfmzNEKnqssFVoNiw84yoiWXbdShMWwIs7N0F4hgVlWjgXspQUNIKorgKmzZkZcYGOAnCmMy4p0FGrT2sgShgCaEMHcCANgYhkEGLW7lXUJYQRwLnfNZB225j3Qe26lpuVDHgJoSCbTuUFBPSgM9Sae0pvfTAOYj7CEQpqdqtCR7WL1o-o2m484ILAtsk3HWPUsRrNcNLKy5Ch7ABfns4C-wSg+msOMf9WyqYYZ4TfX2hYcN9oHZRRUcBUUj1nuIqqNgJyvKhglBxlFl3QJeA4G4fBPBsT4vuMmSAfyUxgGbNix4sC6r0aoOdRjuxkLIvoJc65+MdOUYYXA5CMqkS0s28N-7P3fqI2gOwQaSL3uA9y5SdFShqTuUob2-dR2ejhRTWO3ZeyZgiPZK2sqq3RRsGpCD7SgF+ODmWodUAH0QsAyl1NT69yvs+HWyM5cx09LimQgZmrbI1kHJKbond9XSdjn55KsEYDPWXcpXeHdgLAvLEPWMRZezNvULpV1PW4yxx4uJ4c71yZWpZjAdAGLmuRjti-TQxm4oQBAKOFVkQFmYoFRJ8m0m3jsuFQiTjkqSDuTC+eSzPEk35z-axVLZ4MsvqLDU8qg9h7rogO9oeIlbt1lum+wlkYfstoHkWQWWBGJecVD5os-bYDgGW2m5U64FGlmNLJMyYlaxFmQamm2pzXzPTY8ARMWAiyEw8UIl1RYQCosaL5XSiC0MBDFXxs1x96rZN+4jfThmKbgyRi7OAFzIORhIIrMWuEwH-vtWFSX1BpdkFl3KyMv1yy4GAnwS70DJcxCQDEJXMBxaq-aQ6kgBuYjK7N9At8YBPO8bh7rvT+vDcBheEbm3LELdW499bk3Mu9P28d7ylMjFMzgBd8pS37uc0xE98b03PuFd+56J773ZcHdKFe7DxozRo8S7T5bjPPRgBe8Dyr+X8fS8xHL5n3L2efZh-p2oRnLvq353ceyzxOBosFGzdT-UlJU5PFOH4mA80YDOjzV3of92q2OPtYQIfyWWLj2vc0RES+p8z7sj3gsyvk1r4KI90-z6ssE+nqxpJW7UnB3RT8Rbe4xKPFeL+XhBGwgpOCiHXHVsop-kCl2lrtz04hJ9dkF9T9AlUtiRoESdSF3pARxJEEHJVA7Y1A+cShAQDhjglAsQ7g0AQJ3VFRCY-F35v5FRKC-4AFigT1dQUAONUFgx6DUAmDUhcCfoXwtY91BgwwOpYBM4dYhAp4FwlxbUkYUYbAhCeY+ZugvA+5uhrMiwhgU1ZQtgsw+AoAAAZZ2PcGwW-FJE2MAOTMAKpSgzWH6LOGwZuboQwuAF8CAO2LfIKZ0dwEBN0NZbpc4GwDyYkdWTQwIAAMUTE0F0LeH5jKxwBcHNB4ASDfzeFUHcONz8QQAeGeESLwUIEVnORXmy0H2gigBYXsnrlgEICiJICiO70ETwBDldCiRqTqTgGEUaGgGKNSnZTKJcRiASLUFaWy0jS5CcIUGaBuH-gRnKKwFTmGK32AVnwLGqN73tWnygIsK0JCMMnCP0JmPXHGJwG6BIFaKKJKM6OV1SIgFOktDgK7htAcTuATVgiHxgDCmAUCWKBKCqC0K2MiKmOGARCqHOLIB4ID0GFUJIwKBcC+L0J+ILBTWJG+AMgnCMh4EGjTwTwDz3zrwD03GN3WNCO+NRLj3RO6ExPrxgBxNug6kRPcRABRMtxuBBLgADwAGZugAAWXE4I-E6E1EhkgMJk7oVkmADkykoAA).

e.g.
```ts
console.log(['a', 'b', 1, 'c', 2], getFullLabel(['a', 'b', 1, 'c', 2]))
// ["a", "b", 1, "c", 2],  "c (2)" 
```
in contrary to
```ts
console.log(['d', 'e', 3, 4], getFullLabel(['d', 'e', 3, 4]))
// ["d", "e", 3, 4],  "e (3, 4)" 
```

Should we care about the index of nested objects with arrays?

Reply:
I think we should display the closest label and proper indexes. This feature shouldn't be used in complicated nested forms, though. For those scenarios, inline errors will be more readable.

I'm also unsure about the format of the indexes if this is the best way to display them, but I couldn't find a more readable format.